Extension/Settings.cs

```diff
@@ -22,6 +26,43 @@ public class Settings : SettingsBase
         [XmlElement, SettingProperty(Strings.SETTINGS_ENABLE_AUTO_SORT_NAME, Strings.SETTINGS_ENABLE_AUTO_SORT_TOOLTIP), SettingPropertyGroup(Strings.SETTINGS_GENERAL_GROUP_NAME)]
         public bool AutoSortEnabled { get; set; } = true;
 
+        [XmlElement, SettingProperty(Strings.SETTINGS_RANGED_PREFERENCE_NAME, 0, 2, Strings.SETTINGS_RANGED_PREFERENCE_TOOLTIP), SettingPropertyGroup(Strings.SETTINGS_UPGRADE_GROUP_NAME)]
```

Comment:
```suggestion

#region

#region TroopUpgradingSettings

        [XmlElement, SettingProperty(Strings.SETTINGS_RANGED_PREFERENCE_NAME, 0, 2, Strings.SETTINGS_RANGED_PREFERENCE_TOOLTIP), SettingPropertyGroup(Strings.SETTINGS_UPGRADE_GROUP_NAME)]
```

Reply:
Ive got mixed feelings about using regions you know. I don't like hiding my code since it's still doing something. Maybe we can discuss this later

Reply:
This is a place for very repetitive code which may get rather large if there are more options to come. So I think it makes sense to maintain a region for each settings grouping.